src/go/docker-credential-none/helpers.go

```diff
@@ -0,0 +1,81 @@
+package dcnone
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker-credential-helpers/credentials"
+	dockerconfig "github.com/docker/docker/cli/config"
+)
+
+type dockerConfigType map[string]interface{}
+
+func getParsedConfig() (dockerConfigType, error) {
+	dockerConfig := make(dockerConfigType)
+	contents, err := os.ReadFile(configFile)
+	if err != nil {
+		if errors.Is(err, syscall.ENOENT) {
+			// Time to create a new config (or return no data)
+			return dockerConfig, nil
+		}
+		return dockerConfig, err
+	}
+	err = json.Unmarshal(contents, &dockerConfig)
+	if err != nil {
+		return dockerConfig, fmt.Errorf("reading config file %s: %s", configFile, err)
```

Comment:
Are you sure you want to return `dockerConfig` in this error case rather than `nil`? I don't think I've seen this done before. If `err != nil` here, wouldn't `json.Unmarshal` not modify `dockerConfig`? Either way, it doesn't appear that the first return value is used in the error case anywhere that `getParsedConfig` is called.

Reply:
Good catch. It turns out that if the err return-value is non-nil, none of the callers use the first argument.